backend/backend/routers/api.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=no-member
+
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.exc import IntegrityError
+from sqlmodel import Session, select
+
+from backend.auth import verify_api_key
+from backend.db import get_session
+from backend.models.tables import (
+    Message,
+    MessageCreateAPI,
+    MessagePublic,
+    Order,
+    Provider,
+    ProviderCreate,
+    ProviderPublicWithDetails,
+    User,
+    UserType,
+)
+
+router = APIRouter(prefix="/api", tags=["api"], dependencies=[Depends(verify_api_key)])
+
+
+@router.post("/providers", response_model=ProviderPublicWithDetails, operation_id="apiCreateProvider")
+def create_provider(  # type: ignore
+    provider_payload: ProviderCreate,
+    session: Annotated[Session, Depends(get_session)],
+):
+    sql = select(User).where(
+        User.id.in_(provider_payload.manager_ids),  # type: ignore
+        User.user_type.contains([UserType.PROVIDER_MANAGER]),  # type: ignore
+    )
+
+    db_managers = session.scalars(sql).all()
+
+    if len(db_managers) != len(provider_payload.manager_ids):
+        found_ids = {m.id for m in db_managers}
+        missing_ids = set(provider_payload.manager_ids) - found_ids
+        raise HTTPException(
+            status_code=400, detail=f"Managers {list(missing_ids)} are either missing or not provider managers."
+        )
+
+    db_provider = Provider(name=provider_payload.name, website=provider_payload.website, managers=db_managers)
+
+    try:
+        session.add(db_provider)
+        session.commit()
+        session.refresh(db_provider)
+    except IntegrityError as e:
+        session.rollback()
+        raise HTTPException(status_code=409, detail="Provider already exists") from e
+
+    return db_provider
+
+
+@router.post("/messages", response_model=MessagePublic, operation_id="apiCreateMessage")
+def create_message(  # type: ignore
+    message_payload: MessageCreateAPI,
+    session: Annotated[Session, Depends(get_session)],
+):
+    user = session.get(User, message_payload.user_id)
+    if not user:
+        raise HTTPException(status_code=404, detail=f"User {message_payload.user_id} does not exist.")
+
+    if not UserType.MP_USER in user.user_type:
+        raise HTTPException(status_code=400, detail=f"User {message_payload.user_id} is not an MP_USER.")
+
+    order = session.get(Order, message_payload.order_id)
+    if not order:
+        raise HTTPException(status_code=404, detail=f"Order {message_payload.order_id} does not exist.")
+
+    db_message = Message(author=user, order=order, content=message_payload.content)
+
+    try:
+        session.add(db_message)
+        session.commit()
+        session.refresh(db_message)
+    except Exception as e:
```

Comment:
Can we be narrower here? 
Like `except SQLAlchemyError as e`

It would be nice to include some basic logging 
```
import logging
logger = logging.getLogger(__name__)

logger.exception("Failed to create message")
```